service/frontend/workflow_handler.go

```diff
@@ -4902,16 +4902,16 @@ func validateRequestId(requestID *string, lenLimit int) error {
 func (wh *WorkflowHandler) validateStartWorkflowTimeouts(
 	request *workflowservice.StartWorkflowExecutionRequest,
 ) error {
-	if err := timer.ValidateAndCapTimer(request.GetWorkflowExecutionTimeout()); err != nil {
-		return errInvalidWorkflowExecutionTimeoutSeconds
+	if err := timestamp.ValidateProtoDuration(request.GetWorkflowExecutionTimeout()); err != nil {
+		return fmt.Errorf("%w cause: %s", errInvalidWorkflowExecutionTimeoutSeconds, err.Error())
```

Comment:
can upper layer handle wrapped invalidArgument error?

Reply:
It can. I added checks for error type in the unit tests in `workflow_handler_test` to be sure

Reply:
No it can't. This error goes up as is and eventually returned from handler method, which means it will get converted to status code `Unknown`. You need to do something like:
```suggestion
		return serviceerror.NewInvalidArgument(fmt.Sprintf("%v cause: %v", errInvalidWorkflowExecutionTimeoutSeconds, err)
```
here.

Reply:
I wanted to add helpers for thing like this for a long time. I will do it sooner.

Reply:
and you need to do it everywhere below.

Reply:
yes it can, I added it here: https://github.com/temporalio/api-go/pull/128